tools/probes/probes_main.c

```diff
@@ -219,7 +219,8 @@ void parse_data(char *file_in)
 				/* request to copy full data packet */
 				total_data_to_copy = sizeof(struct probe_data_packet) /
 					sizeof(uint32_t);
-				w_ptr = (uint32_t *)packet;
+				/* probe_data_packet forced to align 1 */
+				w_ptr = __builtin_assume_aligned((uint32_t *)packet, 4);
```

Comment:
can you update the comment too.

Reply:
I think its valid as error shows:
error: converting a packed ‘struct probe_data_packet’ pointer (alignment 1)
it was forced to 1 by 'packed' so we assume it is 4 here

Reply:
@abonislawski but this structure is being read from file into either a buffer that is malloced on at least a 4 byte alignment or the stack (data[]) and is read in chunks of 4 bytes. This is the compiler warning you it's possible the alignment may be 1 byte, but in reality we are telling the compiler that it's 4 bytes based on our mechanism for loading the data. It does not help that we are mixing uint32_t * and struct *, perhaps this needs fixed too.

Reply:
> it's possible the alignment may be 1 byte

Thats why I have added align 1 info in comment, it may be 1 but we know it will not be 1 so assume 4.

If thats wrong please tell me what I should put in comment and lets close this PR

Reply:
4 in comment and 4 in alignment.

Reply:
ok done! :)

Reply:
> This is the compiler warning you it's possible the alignment may be 1 byte,

I think the compiler is only echoing what `packed` ~= `aligned(1)` told it in the first place.

So the current solution seems to be this:
1. `packed`  =>  not aligned
2. `__builtin_assume_aligned(..., 4)` => oh no wait; I changed my mind: actually it is aligned!

What's wrong with `(packed, aligned(4))` and no `__builtin_assume_aligned(..., 4)` ? Fixes the warning for me.

Reply:
BTW here's a very simple demo of the difference between ` probe_data_packet ((packed))` and `probe_data_packet ((packed, aligned(4))`. In this artificial demo, `((packed))` fails the `_Static_assert` whereas `((packed, aligned(4))` doesn't.
```diff
modified   src/probe/probe.c
@@ -64,9 +64,12 @@ struct probe_pdata {
 	struct probe_dma_ext ext_dma;				  /**< extraction DMA */
 	struct probe_dma_ext inject_dma[CONFIG_PROBE_DMA_MAX];	  /**< injection DMA */
 	struct probe_point probe_points[CONFIG_PROBE_POINTS_MAX]; /**< probe points */
+	uint8_t dummy;
 	struct probe_data_packet header;			  /**< data packet header */
 	struct task dmap_work;					  /**< probe task */
 };
+_Static_assert(offsetof(struct probe_pdata, header) % 4 == 0 ,
+               "probe_data_packet is not 4 aligned");
 
 /**
  * \brief Allocate and initialize probe buffer with correct alignment.
```

PS: there are other, unrelated -Werr failures in `probe.c` with `xtensa-apl-elf-gcc (crosstool-NG 1.23.0.440-3b8c) 7.3.0`. I digress.